gdk4-win32/src/win32_display.rs

```diff
@@ -24,4 +25,74 @@ impl Win32Display {
             }
         }
     }
+
+    #[doc(alias = "gdk_win32_display_add_filter")]
+    pub fn add_filter<F>(&self, filter_func: F) -> Win32DisplayFilterHandle
+    where
+        F: Fn(&Self, &mut MSG, &mut i32) -> Win32MessageFilterReturn + 'static,
+    {
+        let box_ = Box::into_raw(Box::new(filter_func)) as *mut _;
+        let func = unsafe {
+            let func: ffi::GdkWin32MessageFilterFunc = Some(trampoline::<F>);
+            ffi::gdk_win32_display_add_filter(self.to_glib_none().0, func, box_);
+            func
+        };
+
+        let display = glib::WeakRef::new();
+        display.set(Some(self));
+
+        let drop_ = |b| unsafe {
+            let _ = Box::<F>::from_raw(b as *mut _);
+        };
+        Win32DisplayFilterHandle {
+            display,
+            func,
+            box_,
+            drop_,
+        }
+    }
+}
+
+unsafe extern "C" fn trampoline<
```

Comment:
The trampoline is only used inside add_filter, so it should be moved into the function itself. That is how we do it everywhere